### PR TITLE
♠️ Move useEthers to context

### PR DESCRIPTION
**Pull request by yivlad**

### CODE REVIEW

File: packages/core/src/providers/network/connectors/context.tsx [hunk 1]
```diff
@@ -32,6 +69,13 @@ export const ConnectorContext = createContext<ConnectorContextValue>({
   //eslint-disable-next-line @typescript-eslint/no-empty-function
   reportError: () => {},
   isLoading: false,
+  setError: () => {
+    throw new Error('Function not implemented.')
+  },
+  active: false,
+  switchNetwork: () => {
+    throw new Error('Function not implemented.')
+  },
```
Comment (sirpy) [inline packages/core/src/providers/network/connectors/context.tsx:78]: should use usecallback
@yivlad

File: packages/core/src/providers/network/connectors/context.tsx [hunk 2]
```diff
@@ -117,23 +196,88 @@ export function ConnectorContextProvider({ children }: ConnectorContextProviderP
...
+        setProvider(controller.getProvider())
+        if (accounts[0]) {
+          setAccount(getAddress(accounts[0]))
+        } else {
+          setAccount(undefined)
+        }
+      }
+      setErrors([...errors])
+    })
+  }, [controller, controller?.getProvider()])
```
Comment (sirpy) [inline packages/core/src/providers/network/connectors/context.tsx:233]: @yivlad missing deps for the set* functions

File: packages/core/src/providers/network/connectors/context.tsx [hunk 3]
```diff
@@ -117,23 +196,88 @@ export function ConnectorContextProvider({ children }: ConnectorContextProviderP
...
+
+    for (const networkState of Object.values(networkStates)) {
+      if (networkState.errors.length > 0) {
+        setError(networkState.errors[networkState.errors.length - 1])
+        return
+      }
+    }
+
+    setError(errors?.[errors.length - 1])
+  }, [chainId, errors, networkStates])
```
Comment (sirpy) [inline packages/core/src/providers/network/connectors/context.tsx:262]: missing deps for seterror

File: packages/core/src/providers/network/connectors/context.tsx (hunk 2 above)
Comment (yivlad) [reply packages/core/src/providers/network/connectors/context.tsx:233]: Not necessary

File: packages/core/src/providers/network/connectors/context.tsx (hunk 3 above)
Comment (yivlad) [reply packages/core/src/providers/network/connectors/context.tsx:262]: Not necessary

File: packages/core/src/providers/network/connectors/context.tsx (hunk 1 above)
Comment (yivlad) [reply packages/core/src/providers/network/connectors/context.tsx:78]: It should not, this value that the context is initiallized with

**Pull request merged**